common/rpc/encryption/localStoreTlsFactory.go

```diff
@@ -241,3 +317,69 @@ func newClientTLSConfig(clientProvider ClientCertProvider, isAuthRequired bool,
 		!clientProvider.DisableHostVerification(isWorker),
 	), nil
 }
+
+func (s *localStoreTlsProvider) timerCallback() {
+	for {
+		select {
+		case <-s.stop:
+			break
+		case <-s.ticker.C:
+		}
+
+		var errorTime time.Time
+		if s.settings.ExpirationChecks.ErrorPeriod != 0 {
+			errorTime = time.Now().UTC().Add(s.settings.ExpirationChecks.ErrorPeriod)
+		} else {
+			errorTime = time.Now().UTC().AddDate(10, 0, 0)
```

Comment:
can this defaults be set at a different layer (perhaps when the ExpirationChecks is being populated?)